### PR TITLE
Frontend improvements

### PR DESCRIPTION
**Pull request by mgorzanski**

### CODE REVIEW

File: src/EpiContentUsage/Frontend/Lib/hooks/useFilteredTableData.ts [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+  ShowColumn = "showColumn",
+  Page = "page",
+  RowsPerPage = "rowsPerPage",
+}
+
+interface FilteredTableDataHookOptions<TableDataType> {
+  rows: TableDataType[];
+  initialTableColumns: TableColumn<TableDataType>[];
+  initialContentTypeBases?: ContentTypeBase[];
+  contentTypeBaseColumnId?: string;
```
Comment (Fifok) [inline src/EpiContentUsage/Frontend/Lib/hooks/useFilteredTableData.ts:21]: Can contentTypeBaseColumnId be different than FilteredTableDataQueryParam.ContentTypeBase? Maybe it's worth to change string to FilteredTableDataQueryParam.ContentTypeBase.
```suggestion
  contentTypeBaseColumnId?: FilteredTableDataQueryParam.ContentTypeBase;
```

File: src/Foundation/Startup.cs [hunk 2]
```diff
@@ -228,7 +228,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             app.UseAnonymousId();
             app.UseStaticFiles();
             app.UseRouting();
-            app.UseCors();
+            app.UseCors(options => options.WithOrigins("http://localhost:8080").AllowAnyMethod().AllowAnyHeader().AllowAnyOrigin());
```
Comment (Fifok) [inline src/Foundation/Startup.cs:231]: It should be only enabled when env.IsDevelopment() is true.

File: src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx [hunk 3]
```diff
@@ -1,441 +1,448 @@
...
+  tableColumns: TableColumn<TableDataType>[];
+  onRowClick?: (event: React.PointerEvent) => void;
+  onEditButtonClick?: (event: React.PointerEvent) => void;
+  onViewWebsiteClick?: () => void;
+  hasDiscloseTableRows?: boolean;
+}
+
+const ContentTypeUsageDiscloseTableRow = ({
+  tableColumns,
+  onRowClick,
```
Comment (Fifok) [inline src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx:49]: Never used

File: src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx [hunk 4]
```diff
@@ -1,441 +1,448 @@
...
+  ...row
+}: ContentTypeUsageTableRowProps<ContentUsageDto>) => {
+  const {
+    views: {
+      contentUsagesView: {
+        table: { actions },
```
Comment (Fifok) [inline src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx:55]: actions is never used

File: src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx [hunk 5]
```diff
@@ -1,441 +1,448 @@
...
-const ROWS_PER_PAGE_OPTIONS = [15, 30, 60];
+interface ContentTypeUsageTableRowProps<TableDataType> extends ContentUsageDto {
+  tableColumns: TableColumn<TableDataType>[];
+  onRowClick?: (event: React.PointerEvent) => void;
+  onEditButtonClick?: (event: React.PointerEvent) => void;
+  onViewWebsiteClick?: () => void;
+  hasDiscloseTableRows?: boolean;
+}
+
+const ContentTypeUsageDiscloseTableRow = ({
```
Comment (Fifok) [inline src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx:47]: Disclose row doesn't display action buttons. I propose to create new issue for this.

File: src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx (hunk 3 above)
Comment (mgorzanski) [reply src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx:49]: Yes..., but will be used in the future ;)

File: src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx (hunk 4 above)
Comment (mgorzanski) [reply src/EpiContentUsage/Frontend/Views/ContentTypeUsageView.tsx:55]: The same as above

**Review by Fifok — APPROVED** [review]

**Pull request merged**